src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -134,5 +127,268 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
+                    WellKnownMember.System_Runtime_CompilerServices_InterpolatedStringBuilder__Dispose,
+                    diagnostics,
+                    syntax: unconvertedInterpolatedString.Syntax);
+
+                MethodArgumentInfo? disposeInfo = null;
+
+                if (disposeMethod != null)
+                {
+                    disposeInfo = new MethodArgumentInfo(disposeMethod, argumentsBuilder.ToImmutable(), ArgsToParamsOpt: default, DefaultArguments: default, Expanded: false);
+
+                    ReportDiagnosticsIfObsolete(diagnostics, disposeMethod, unconvertedInterpolatedString.Syntax, hasBaseReceiver: false);
+                    ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, disposeMethod, unconvertedInterpolatedString.Syntax.Location, isDelegateConversion: false);
+                }
+
+                argumentsBuilder.Free();
+                result = constructWithData(
+                    newParts,
+                    new InterpolatedStringBuilderData(
+                        interpolatedStringBuilderType,
+                        constructorInfo,
+                        interpolatedDataBuilder.ToImmutableAndFree(),
+                        disposeInfo,
+                        LocalScopeDepth));
+                return true;
+            }
+        }
+
+        private ImmutableArray<BoundExpression> BindInterpolatedStringParts(BoundUnconvertedInterpolatedString unconvertedInterpolatedString, BindingDiagnosticBag diagnostics)
+        {
+            ArrayBuilder<BoundExpression>? partsBuilder = null;
+            var objectType = GetSpecialType(SpecialType.System_Object, diagnostics, unconvertedInterpolatedString.Syntax);
+            for (int i = 0; i < unconvertedInterpolatedString.Parts.Length; i++)
+            {
+                var part = unconvertedInterpolatedString.Parts[i];
+                if (part is BoundStringInsert insert)
+                {
+                    BoundExpression newValue;
+                    if (insert.Value.Type is null)
+                    {
+                        newValue = GenerateConversionForAssignment(objectType, insert.Value, diagnostics);
+                    }
+                    else
+                    {
+                        newValue = BindToNaturalType(insert.Value, diagnostics);
+                        _ = GenerateConversionForAssignment(objectType, insert.Value, diagnostics);
+                    }
+
+                    if (insert.Value != newValue)
+                    {
+                        if (partsBuilder is null)
+                        {
+                            partsBuilder = ArrayBuilder<BoundExpression>.GetInstance(unconvertedInterpolatedString.Parts.Length);
+                            partsBuilder.AddRange(unconvertedInterpolatedString.Parts, i);
+                        }
+
+                        partsBuilder.Add(insert.Update(newValue, insert.Alignment, insert.Format, insert.Type));
```

Comment:
>insert.Type [](start = 98, length = 11)

It looks like ```BoundStringInsert.Type``` is always null. Consider adjusting the node definition accordingly so that we wouldn't need to pass the type and there would be no questions if the type should also be adjusted here and in other similar places.